pkg/kubectl/rollout_status.go

```diff
@@ -107,3 +114,34 @@ func (s *DaemonSetStatusViewer) Status(namespace, name string, revision int64) (
 	}
 	return fmt.Sprintf("Waiting for daemon set spec update to be observed...\n"), false, nil
 }
+
+// Status returns a message describing statefulset status, and a bool value indicating if the status is considered done
+func (s *StatefulSetStatusViewer) Status(namespace, name string, revision int64) (string, bool, error) {
+	sts, err := s.c.StatefulSets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", false, err
+	}
+	if sts.Spec.UpdateStrategy.Type == apps.OnDeleteStatefulSetStrategyType {
+		return "", true, fmt.Errorf("%s updateStrategy does not have a Status`", apps.OnDeleteStatefulSetStrategyType)
+	}
+	if sts.Status.ObservedGeneration == nil || sts.Generation > *sts.Status.ObservedGeneration {
+		return "Waiting for statefulset spec update to be observed...\n", false, nil
+	}
+	if sts.Status.ReadyReplicas < sts.Spec.Replicas {
+		return fmt.Sprintf("Waiting for %d pods to be ready...\n", sts.Spec.Replicas-sts.Status.ReadyReplicas), false, nil
+	}
+	if sts.Spec.UpdateStrategy.Type == apps.PartitionStatefulSetStrategyType {
+		if sts.Status.UpdatedReplicas < (sts.Spec.Replicas - sts.Spec.UpdateStrategy.Partition.Ordinal) {
+			return fmt.Sprintf("Waiting for partitioned roll out to finish: %d out of %d new pods have been updated...\n",
+				sts.Status.UpdatedReplicas, (sts.Spec.Replicas - sts.Spec.UpdateStrategy.Partition.Ordinal)), false, nil
+		}
+		return fmt.Sprintf("partitioned roll out complete: %d new pods have been updated...\n",
+			sts.Status.UpdatedReplicas), false, nil
+	}
+	if sts.Status.UpdateRevision != "" {
```

Comment:
You probably want to update this one too as part of moving away from resetting updateRevision and updatedReplicas.